### PR TITLE
chore(hybrid-cloud): Update team notifications settings details endpoint for split silo mode

### PR DESCRIPTION
**Pull request by dashed**

### CODE REVIEW

File: src/sentry/services/hybrid_cloud/notifications/impl.py [hunk 1]
```diff
@@ -271,10 +295,20 @@ def apply_filters(
...
-                query = query.filter(team_id__in=filters["team_ids"])
+            if (
+                "user_ids" in filters
+                and "team_ids" in filters
+                and len(filters["user_ids"]) > 0
+                and len(filters["team_ids"]) > 0
+            ):
+                query = query.filter(
+                    Q(user_id__in=filters["user_ids"]) | Q(team_id__in=filters["team_ids"])
+                )
```
Comment (dashed) [inline src/sentry/services/hybrid_cloud/notifications/impl.py:306]: This matches the behaviour of `def _filter()` as implemented here https://github.com/getsentry/sentry/blob/e8c2b619393f3d7667473ba4f1db373fd65cb514/src/sentry/notifications/manager.py#L306-L310

**Review by scefali — COMMENTED** [review]
@dashed I don't think we even use these endpoints. Can we just delete these endpoints instead?

**Comment by corps** [comment]
> @dashed I don't think we even use these endpoints. Can we just delete these endpoints instead?

<3 <3 <3

**Comment by mrduncan** [comment]
Confirmed no requests to this endpoint in the past 7 days in SaaS - should be safe to remove

**Comment by RyanSkonnord** [comment]
Superseded by https://github.com/getsentry/sentry/pull/59371

**Pull request closed without merging**